src/Repository/NewsletterSubscriptionRepository.php

```diff
@@ -77,6 +77,26 @@ public function findOneByEmail(string $email): ?NewsletterSubscription
         return $this->findOneBy(['email' => $email]);
     }
 
+    public function findOneNotConfirmedByEmail(string $email): ?NewsletterSubscription
+    {
+        return $this
+            ->createQueryBuilder('newsletter')
+            ->where('newsletter.email = :email')
+            ->andWhere('newsletter.confirmedAt IS NULL')
```

Comment:
just asking : when someone unsubscribe from the newsletter, is this field set to null ? If not, someone who has unsubscribe will not be able to re-subscribe, right ?

Reply:
It's still possible to subscribe, because to find soft deleted subscription we use only its mail.
But good point about `confirmedAt`!
Now I set `confirmedAt` to NULL when we recover soft deleted subscription.